ksql-api/src/main/java/io/confluent/ksql/api/server/BasePublisher.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import java.util.Objects;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Base class for our reactive streams publishers
+ *
+ * @param <T> the type of the element
+ */
+public abstract class BasePublisher<T> implements Publisher<T> {
+
+  private static final Logger log = LoggerFactory.getLogger(BasePublisher.class);
+
+  protected final Context ctx;
+  protected Subscriber<? super T> subscriber;
+  protected long demand;
+  protected boolean cancelled;
+
+  public BasePublisher(final Context ctx) {
+    this.ctx = ctx;
```

Comment:
nit: validate params that will be stored in object state; ensuring object does not get into an invalid state.